src/core/jpg.js

```diff
@@ -332,6 +332,7 @@ var JpegImage = (function JpegImageClosure() {
     if (!resetInterval) {
       resetInterval = mcuExpected;
     }
+    var nonDefaultResetInterval = resetInterval !== mcuExpected;
```

Comment:
duh. how about `var toRead = resetInterval ? Math.min(mcuExpected - mcu, resetInterval) : mcuExpected;` inside the loop?